msal/application.py

```diff
@@ -1841,6 +1845,7 @@ def acquire_token_by_username_password(
             - A successful response would contain "access_token" key,
             - an error response would contain "error" and usually "error_description".
         """
+        warnings.warn("This API has been deprecated, please use a more secure flow. See https://aka.ms/msal-ropc-migration for migration guidance", DeprecationWarning)
```

Comment:
1. First of all, the ROPC flow was available for both public client application (PCA) and confidential client application (CCA). The work item in PR description does not provide any detail on its scope, either. Assuming this PR means for PCA only (because the [ROPC deprecation in CCA is still TBD](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/5366)), we must add this warning conditionally.
2. [Nitpick] The [line length is recommended to be less than 79 characters](https://peps.python.org/pep-0008/#maximum-line-length). You may use the multi-line string technique here.

```suggestion
        warnings.warn("""This API has been deprecated, please use a more secure flow.
See https://aka.ms/msal-ropc-migration for migration guidance""", DeprecationWarning)
```
3. In fact, one `warnings.warn(...)` in this code path is the only needed treatment in this PR. We shall revert all other changes in this PR because, until we actually remove ropc in the next major version bump, the feature needs to exist, and all the tests are therefore still useful.

Reply:
Understood! I've made these changes